iquip/apps/logger.py

```diff
@@ -0,0 +1,188 @@
+        # layouts
+        layout = QVBoxLayout(self)
+        layout.addWidget(self.label)
+        layout.addWidget(self.buttonBox)
+
+    def buttonOKClicked(self):
+        """Called when the OK button is clicked."""
+        self.confirmed.emit()
+        self.close()
+
+    def buttonCancelClicked(self):
+        """Called when the Cancel button is clicked."""
+        self.close()
+
+
+class LoggerApp(qiwis.BaseApp):
+    """App for logging.
+
+    Manages a logger frame.
+
+    Attributes:
+        loggerFrame: A frame that shows the logs.
+        confirmFame: A frame that asks whether to clear logs.
+        handler: A handler for adding logs to GUI. 
+    """
+
+    def __init__(self, name: str, parent: Optional[QObject] = None):
+        """Extended.
+
+        Args:
+            name: Name of the App
+        """
+        super().__init__(name, parent=parent)
+        self.loggerFrame = LoggerFrame()
+        # connect signals to slots
+        self.loggerFrame.clearButton.clicked.connect(self.checkToClear)
+        self.confirmFrame = ConfirmClearingFrame()
+        self.confirmFrame.confirmed.connect(self.clearLog)
+        self.handler = LoggingHandler(self.addLog)
+        # TODO(aijuh): Change the log format when it is determined.
+        fs ="%(levelname)s [%(name)s] [%(filename)s:%(lineno)d] %(message)s "
+        formatter = logging.Formatter(fs)
+        self.handler.setFormatter(formatter)
+        rootLogger = logging.getLogger()
+        rootLogger.setLevel(logging.DEBUG)
+        rootLogger.addHandler(self.handler)
+        self.handler.setLevel(logging.WARNING)
+        self.loggerFrame.levelBox.addItems(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"])
+        self.loggerFrame.levelBox.textActivated.connect(self.setLevel)
+        self.loggerFrame.levelBox.setCurrentText("WARNING")
+
+    @pyqtSlot(str)
+    def setLevel(self, text: str):
+        """Responds to the setLevelBox widget and changes the handler's level.
+
+        Args:
+            text: Selected level in the level select box.
+              It should be one of "DEBUG", "INFO", "WARNING", "ERROR" and "CRITICAL".
+              It should be case-sensitive and any other input is ignored.
+        """
+        level = {
+            "DEBUG": logging.DEBUG,
+            "INFO": logging.INFO,
+            "WARNING": logging.WARNING,
+            "ERROR": logging.ERROR,
+            "CRITICAL": logging.CRITICAL
+        }
+        if text in level:
+            self.handler.setLevel(level[text])
+            logging.getLogger().setLevel(level[text])
+
+    def frames(self) -> Tuple[LoggerFrame]:
+        """Overridden."""
+        return (self.loggerFrame,)
+
+    @pyqtSlot(str)
+    def addLog(self, content: str):
+        """Adds a channel name and log message.
+
+        Args:
+            content: Received log message.
+        """
+        timeString = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime(time.time()))
```

Comment:
It is just an option, and you can use `QDateTime.currentDateTime()` to obtain the current time.
I used this in `builder.py` and it may look more unified.